src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -246,7 +246,9 @@ trait LowPriorityConfigConvertImplicits {
 
   implicit val readString = fromString[String](identity)
   implicit val readBoolean = fromString[Boolean](_.toBoolean)
-  implicit val readDouble = fromString[Double](_.toDouble)
+  implicit val readDouble = fromString[Double]({
+    v => if (v.last == '%') v.dropRight(1).toDouble / 100.0 else v.toDouble
```

Comment:
Maybe we could group the code for non empty strings together and provide it as utility to build instances. Something like:

```scala
def fromNonEmptyString[T](name: String, fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
    override def from(config: ConfigValue): Try[T] = fromFConvert { _ match {
      case "" => throw new Exception(s"Cannot read a $name from an empty string")
      case x => fromF(x)
    }(config)
    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
}
```

It would remove some code and group together the login for empty String errors. What do you think?